Stage feature branch

mckarthik7 opened the pull request:
Something that I had in my local repo for some time now to make plagiarism testing easier

 - Created option to make a 'stage' folder under netid with the submission that is tested
 - Print command string for moss to use easily with this stage option

Details in commit messages. Feel free to merge

Commit:
grader.py: Added option to make a permanent staging directory

Added option that creates a permanent staging directory under
<submit_path>/<netid>/stage/ . It simply copies over the valid
submission files used for grading. This is useful to identify the
submission that is used for each student so it can be used easily with
plagiarism scripts like moss

- Created new binary CLI argument called stage_submission to enable this
- Created new function `setup_perm_stage` that only copies over
  submitted files
- Execution of each phase still happens in tmp to avoid running user
  code in stage directory and clean up easily

Commit:
grader.py: Added printing moss command string for testing submission

Added printing moss command string that can be used to directly run the
plagiarism test with correct arguments.
- script sets the base template files
- script sets directory option so it doesn't check files in the same
  directory (i.e. same student submission)
- script doesn't set language option. Use c for C, cc for CPP.

RichardHabeeb commented on grader/grader.py:
```diff
@@ -329,9 +329,15 @@ def main(gradebook_input_file,
-        print("[i] Updating gradebook (" + gradebook_output_path + ")")
-        book.write_grades(gradebook_output_path)
-
+    print("[i] Updating gradebook (" + gradebook_output_path + ")")
```
Can we leave these lines indented? This way if you need to partially stop the script it can resume where it left off.

RichardHabeeb reviewed (changes requested):
Can we leave the

Commit:
grader.py: indent print and book.write into for loop

indent print and book.write statements into the for loop as suggested in
PR. This can be used to partially stop the script and start again from
where we left off
